### PR TITLE
implement performance mode for existing state size check

### LINKED ISSUE #4388: add reflex perf mode
add rx.Config (perf_mode) or env variable (REFLEX_PERF_MODE). Value should be an enum with warn, raise and ignore. Default should be warn, to avoid breaking changes.

warn: Warn about possible performance implications
raise: raise Exception if performance issues are found
ignore: do nothing, disable checks completely

iirc reflex currently only checks for state sizes. I would love to add another check for always dirty substates (caused by uncached vars).

### PR DESCRIPTION
**Pull request by benedikt-bartscher**
Wip #4388

Bonus: made state size warning adjustable via env. Currently, it is hardcoded to 100kb

### CODE REVIEW

**Review by masenf — COMMENTED** [review]
nice, seems good to me

File: reflex/config.py [hunk 1]
```diff
@@ -545,6 +553,12 @@ class EnvironmentVariables:
     # Where to save screenshots when tests fail.
     SCREENSHOT_DIR: EnvVar[Optional[Path]] = env_var(None)
 
+    # In which performance mode to run the app.
+    REFLEX_PERF_MODE: EnvVar[Optional[PerformanceMode]] = env_var(PerformanceMode.WARN)
+
+    # The maximum size of the reflex state in kilobytes.
+    REFLEX_STATE_SIZE_LIMIT: EnvVar[int] = env_var(1000)
```
[the remark this answers is not in the document]
Comment (benedikt-bartscher) [reply reflex/config.py:560]: I didn't even know about a ws size limit. I will take a look at this

[the remark this answers is not in the document]
Comment (benedikt-bartscher) [reply reflex/config.py:560]: @Lendemor where do I find the ws limit?

**Comment by benedikt-bartscher** [comment]
Let's merge this as-is and continue further checks in a followup?

File: reflex/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (masenf) [reply reflex/config.py:560]: it's 1mb from socketio https://python-socketio.readthedocs.io/en/stable/api.html

> max_http_buffer_size – The maximum size that is accepted for incoming messages. The default is 1,000,000 bytes. In spite of its name, the value set in this argument is enforced for HTTP long-polling and WebSocket connections.

**Comment by adhami3310** [comment]
@benedikt-bartscher we need a rebase for this :D

**Comment by benedikt-bartscher** [comment]
> @benedikt-bartscher we need a rebase for this :D

Thanks for the heads-up, done :heavy_check_mark:

**Review by masenf — APPROVED** [review]

**Pull request merged**